kaldb/src/test/java/com/slack/kaldb/testlib/KaldbGrpcQueryUtil.java

```diff
@@ -0,0 +1,78 @@
+package com.slack.kaldb.testlib;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linecorp.armeria.client.Clients;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import com.slack.kaldb.proto.service.KaldbSearch;
+import com.slack.kaldb.proto.service.KaldbServiceGrpc;
+import java.io.IOException;
+import java.util.HashMap;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KaldbGrpcQueryUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(KaldbGrpcQueryUtil.class);
+
+  public static KaldbSearch.SearchResult searchUsingGrpcApi(
```

Comment:
This should probably remain part of the indexer test and not extracted to a general purpose utility. 

This is taking a builder pattern and puts it into a method with a significant amount of arguments. This makes reading the code much more difficult as instead of having self-documenting method calls you just have a list of values.

```java
searchUsingGrpcApi(
            "*:*", 0L, 1601547099000L, MessageUtil.TEST_INDEX_NAME, 100, 2, queryServicePort + 1);
```

```java
kaldbService.search(
        KaldbSearch.SearchRequest.newBuilder()
            .setIndexName(MessageUtil.TEST_INDEX_NAME)
            .setQueryString("*:*")
            .setStartTimeEpochMs(0L)
            .setEndTimeEpochMs(1601547099000L)
            .setHowMany(100)
            .setBucketCount(2)
            .build());
```

Reply:
Moved this code into KaldbTest. Reduced the number of fields passed in also.